api/client/info.go

```diff
@@ -19,127 +19,76 @@ func (cli *DockerCli) CmdInfo(args ...string) error {
 	cmd.Require(flag.Exact, 0)
 	cmd.ParseFlags(args, false)
 
-	body, _, err := readBody(cli.call("GET", "/info", nil, nil))
+	rdr, _, err := cli.call("GET", "/info", nil, nil)
 	if err != nil {
 		return err
 	}
 
-	out := engine.NewOutput()
-	remoteInfo, err := out.AddEnv()
-	if err != nil {
-		return err
-	}
-
-	if _, err := out.Write(body); err != nil {
-		logrus.Errorf("Error reading remote info: %s", err)
+	info := &types.Info{}
+	if err := json.NewDecoder(rdr).Decode(info); err != nil {
+		logrus.Errorf("Error reading remote info: %v", err)
```

Comment:
I don't think we need this error message here since we're returning error below.
Maybe update the error with this extra info, though?